esp32-wroom-rp/src/tcp_client.rs

```diff
@@ -8,28 +9,51 @@ use super::network::{IpAddress, Socket};
 
 use embedded_hal::blocking::spi::Transfer;
 
-pub struct TcpClient<'a, 'b, B, C> {
-    pub(crate) protocol_handler: &'a mut NinaProtocolHandler<'a, B, C>,
+pub struct TcpClient<'a, B, C> {
+    pub(crate) protocol_handler: &'a mut NinaProtocolHandler<B, C>,
+    pub(crate) socket: Option<Socket>,
     pub(crate) server_ip_address: Option<IpAddress>,
-    pub(crate) server_hostname: Option<&'b str>,
+    pub(crate) server_hostname: Option<&'a str>,
 }
 
-impl<'a, 'b, B, C> TcpClient<'a, 'b, B, C>
+impl<'a, B, C> TcpClient<'a, B, C>
 where
     B: Transfer<u8>,
     C: EspControlInterface,
 {
-    fn server_ip_address(mut self, ip: IpAddress) -> Self {
+    pub fn build(wifi: &'a mut Wifi<B, C>) -> Self {
+        Self {
+            protocol_handler: wifi.protocol_handler.get_mut(),
+            socket: None,
+            server_ip_address: None,
+            server_hostname: None,
+        }
+    }
+
+    pub fn connect<F>(mut self, ip: IpAddress, f: F)
+    where
+        F: Fn(TcpClient<'a, B, C>),
+    {
+        let socket = self.get_socket().unwrap();
```

Comment:
Something curious, why is this `unwrap()` needed when the getter has an inline `unwrap()`?